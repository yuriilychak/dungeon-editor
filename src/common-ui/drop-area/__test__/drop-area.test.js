import React from "react";
import { createMount } from "@material-ui/core/test-utils";
import DropArea from "../drop-area";

describe("DropArea test",()=> {

    const dropMessage = "Test";
    const onDropFiles = jest.fn();

    it("default snapshot", () => {
        const wrapper = createMount()(
            <DropArea
                dropMessage={dropMessage}
                onDropFiles={onDropFiles}
            />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("with children snapshot", () => {
        const wrapper = createMount()(
            <DropArea
                dropMessage={dropMessage}
                onDropFiles={onDropFiles}
            >
                Children
            </DropArea>
        );
        expect(wrapper.html()).toMatchSnapshot();
    });
});
