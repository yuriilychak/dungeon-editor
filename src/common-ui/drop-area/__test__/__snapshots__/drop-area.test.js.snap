// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropArea test default snapshot 1`] = `"<div class=\\"drop-area-root\\"><input multiple=\\"\\" type=\\"file\\" style=\\"display: none;\\" autocomplete=\\"off\\" tabindex=\\"-1\\"><div class=\\"drop-area-overflow\\"></div><div class=\\"drop-area-drag\\" style=\\"opacity: 0; visibility: hidden;\\">Test</div></div>"`;

exports[`DropArea test with children snapshot 1`] = `"<div class=\\"drop-area-root\\"><input multiple=\\"\\" type=\\"file\\" style=\\"display: none;\\" autocomplete=\\"off\\" tabindex=\\"-1\\"><div class=\\"drop-area-overflow\\">Children</div><div class=\\"drop-area-drag\\" style=\\"opacity: 0; visibility: hidden;\\">Test</div></div>"`;
